### ISSUE TITLE
[Feature Request]: Rewrite pytest markers to avoid including ML test suites into a catch-all poscommit suite.

### ISSUE THREAD

**Issue by tvalentyn** [apache/beam#28578]
### What would you like to happen?

Current test configuration causes confusion for authors of new suites and makes some test suites run twice (see discussions on: https://github.com/apache/beam/pull/28571 https://github.com/apache/beam/pull/28534)

### Issue Priority

Priority: 3 (nice-to-have improvement)

### Issue Components

- [X] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

**Comment by AnandInguva** [comment]
if a test is using `it_postcommit` along with a ML framework marker, it is considered to be an IT test. These IT tests then will be run on Dataflow with the respective requirements file. We provide both the markers because we want to only collect IT tests of that ML framework but not unit tests.

**Comment by damccorm** [comment]
I don't think this should be a release blocker. The problem in question is that tests are running more often than they should, which doesn't represent risk to users.

@tvalentyn I'm removing the release blocker, if you disagree feel free to add it back
